### PR TITLE
Cleanup App Engine samples.

### PR DESCRIPTION
**Pull request by jmdobry**
## App Engine samples

- Deleted
  - [x] - `disk`
  - [x] - `logging`
- Redirected to https://cloud.google.com/community/tutorials
  - [x] - `bower`
  - [x] - `express`
  - [x] - `express-memcached-session`
  - [x] - `extending-runtime`
  - [x] - `geddy`
  - [x] - `hapi`
  - [x] - `koa`
  - [x] - `kraken`
  - [x] - `mailgun`
  - [x] - `restify`
  - [x] - `sails`
  - [x] - `webpack`
- Updated
  - [x] - `analytics`
  - [x] - `cloudsql`
  - [x] - `datastore`
  - [ ] - `endpoints`
  - [ ] - `errorreporting`
  - [ ] - ~`grunt`~
  - [x] - `hello-world`
  - [ ] - ~`loopback`~
  - [ ] - `mailjet`
  - [ ] - `memcached`
  - [ ] - `mongodb`
  - [ ] - `parse-server`
  - [ ] - `pubsub`
  - [ ] - `redis`
  - [ ] - `sendgrid`
  - [ ] - `static-files`
  - [ ] - `storage`
  - [ ] - `twilio`
  - [ ] - `websockets`
  
## Other samples

- [x] - `bigquery`
- [x] - `computeengine`
- [x] - `containerengine/hello-world`
- [x] - `datastore`
- [x] - `debugger`
- [x] - `dlp`
- [x] - `dns`
- [x] - `endpoints`
- [x] - `functions`
- [x] - `kms`
- [x] - `language`
- [x] - `logging`
- [x] - `monitoring`
- [x] - `prediction`
- [x] - `pubsub`
- [x] - `resource`
- [x] - `spanner`
- [x] - `speech`
- [x] - `storage`
- [x] - `trace`
- [x] - `translate`
- [x] - `video`
- [x] - `vision`

### CODE REVIEW

File: resource/system-test/projects.test.js [hunk 1]
```diff
@@ -15,15 +15,15 @@
...
 const cwd = path.join(__dirname, `..`);
 const cmd = `node projects.js`;
 
 test(`should list projects`, async (t) => {
-  const stdout = await runAsync(`${cmd} list`, cwd);
-  t.true(stdout.includes(`Projects:`));
-  t.true(stdout.includes(`${process.env.GCLOUD_PROJECT}`));
+  const output = await tools.runAsync(`${cmd} list`, cwd);
+  t.true(output.includes(`Projects:`));
+  t.true(output.includes(`${process.env.GCLOUD_PROJECT}`));
```
Comment (ace-n) [inline resource/system-test/projects.test.js]: I assume we're checking to make sure `GCLOUD_PROJECT` is set before this?

File: speech/system-test/quickstart.test.js [hunk 2]
```diff
@@ -15,11 +15,17 @@
...
 
-require(`../../system-test/_setup`);
-
 const path = require(`path`);
 const proxyquire = require(`proxyquire`).noPreserveCache();
+const sinon = require(`sinon`);
 const speech = proxyquire(`@google-cloud/speech`, {})();
+const test = require(`ava`);
+
+const {
```
Comment (ace-n) [inline speech/system-test/quickstart.test.js]: I don't think Node v4 supports this format.

(Caveat: I think we may have talked about only supporting Node v6 in our tests, but I can't recall. If that's what we do, there's a [small] risk our samples may pass on Node v6 but fail on Node v4.)

File: storage/transfer.js [hunk 3]
```diff
@@ -348,7 +348,6 @@ function resumeTransferOperation (transferOperationName, callback) {
 
 // The command-line program
 var cli = require('yargs');
```
Comment (ace-n) [inline storage/transfer.js]: Nit for another PR: use `const` instead?

File: functions/pubsub/package.json [hunk 4]
```diff
@@ -2,13 +2,29 @@
...
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/GoogleCloudPlatform/nodejs-docs-samples.git"
+  },
+  "cloud": {
+    "requiresKeyFile": true,
+    "requiresProjectId": true
+  },
+  "engines": {
+    "node": ">=4.3.2"
```
Comment (jmdobry) [inline functions/pubsub/package.json]: good question

File: dns/zones.js [hunk 5]
```diff
@@ -15,43 +15,34 @@
...
-const program = module.exports = {
-  listZones: listZones,
-  main: (args) => {
-    // Run the command-line program
-    cli.help().strict().parse(args).argv;
-  }
-};
-
-cli
+require(`yargs`) // eslint-disable-line
```
Comment (jmdobry) [inline dns/zones.js]: In this case I think the linter is lame.

File: logging/system-test/logs.test.js [hunk 6]
```diff
@@ -44,26 +48,17 @@ test.cb.serial(`should write a log entry`, (t) => {
...
+    const entries = await program.listLogEntriesAdvanced(filter, 5, null);
+    assert(entries.some((entry) => entry.data && entry.data.message === message));
   }).start();
 });
 
 test.cb.serial(`should delete a log`, (t) => {
   program.deleteLog(logName, (err, apiResponse) => {
     // Ignore "Not Found" error
-    if (err && err.code !== 404) {
+    if (err && err.code !== 5) {
```
Comment (jmdobry) [inline logging/system-test/logs.test.js]: Yes, the library moved to grpc

File: resource/system-test/projects.test.js (hunk 1 above)
Comment (jmdobry) [inline resource/system-test/projects.test.js]: Fixed.

File: speech/system-test/quickstart.test.js (hunk 2 above)
Comment (jmdobry) [inline speech/system-test/quickstart.test.js]: The tests are transpiled by Ava. Our use of async/await in our tests definitely doesn't work in Node v4.

File: storage/transfer.js (hunk 3 above)
Comment (jmdobry) [inline storage/transfer.js]: Another PR.

File: dlp/package.json [hunk 7]
```diff
@@ -3,38 +3,33 @@
...
     "node": ">=4.3.2"
   },
+  "scripts": {
+    "system-test": "ava -T 1m --verbose system-test/*.test.js",
+    "test": "npm run system-test"
+  },
   "dependencies": {
-    "google-auth-library": "^0.10.0",
-    "google-auto-auth": "^0.5.2",
+    "google-auth-library": "0.10.0",
```
Comment (jmdobry) [inline dlp/package.json]: I'm removing it everywhere, as we're moving towards automating dependency updates. We can pin to versions, and let the automation update them. When you pin versions it's easier to get reproducible builds and know what version you have installed.

File: datastore/package.json [hunk 8]
```diff
@@ -2,16 +2,31 @@
...
-  "scripts": {
-    "test": "cd ..; npm run st -- --verbose datastore/system-test/*.test.js"
+  "repository": {
+    "type": "git",
+    "url": "https://github.com/GoogleCloudPlatform/nodejs-docs-samples.git"
   },
-  "dependencies": {
-    "@google-cloud/datastore": "0.8.0",
-    "yargs": "6.6.0"
+  "cloud": {
```
Comment (jmdobry) [inline datastore/package.json]: good point

**Pull request merged**